core/coreapi/interface/interface.go

```diff
@@ -38,6 +40,12 @@ type UnixfsAPI interface {
 	Ls(context.Context, Path) ([]*Link, error)
 }
 
+type DagAPI interface {
+	Add(ctx context.Context, src io.Reader, inputEnc string, format cid.Prefix) ([]Node, error)
```

Comment:
Maybe `Put` instead of `Add`? 

I also think its important for people to be able to call these methods without having to import other packages. So making the `format` argument a pointer, and having some sort of defaults for when the user passes `nil` is something we should think about.